packages/skills/weather_client_ledger/handlers.py

```diff
@@ -206,20 +210,22 @@ def _handle_match_accept(self, msg: FIPAMessage, sender: str, message_id: int, d
         logger.info("[{}]: received MATCH_ACCEPT_W_ADDRESS from sender={}".format(self.context.agent_name, sender[-5:]))
         address = cast(str, msg.get("address"))
         proposal = cast(Description, dialogue.proposal)
+        crypto = CURRENCY_TO_ID_MAP[proposal.values.get("currency_pbk")]
         tx_msg = TransactionMessage(performative=TransactionMessage.Performative.PROPOSE,
                                     skill_id="weather_client_ledger",
                                     transaction_id="transaction0",
-                                    sender=self.context.agent_public_keys['fetchai'],
+                                    sender=self.context.agent_public_keys[crypto],
                                     counterparty=address,
                                     is_sender_buyer=True,
-                                    currency_pbk="FET",
+                                    currency_pbk=crypto,
```

Comment:
the currency_pbk should be the `currency_pbk` not the `ledger_id`

Reply:
This is based on the logic you had on the decision_maker. You were choosing which address to use based on the currency_pbk.